pkg/master/master.go

```diff
@@ -588,8 +638,8 @@ func (m *Master) getExtensionResources(c *Config) map[string]rest.Storage {
 	}
 
 	storage := map[string]rest.Storage{}
-	if isEnabled("horizontalpodautoscalers") {
-		autoscalerStorage, autoscalerStatusStorage := horizontalpodautoscaleretcd.NewREST(dbClient("horizontalpodautoscalers"), storageDecorator)
+	// Install only when appriopriate storages are provided.
+	if isEnabled("horizontalpodautoscalers") && autoscalerStorage != nil && autoscalerStatusStorage != nil {
 		storage["horizontalpodautoscalers"] = autoscalerStorage
 		storage["horizontalpodautoscalers/status"] = autoscalerStatusStorage
```

Comment:
Not sure what do you mean. If you are afraid that querying api server for extensions/v1beta1/hpa will return object in version autoscaling/v1 this is not true. Also I checked manually it:
```
curl localhost:8080/apis/extensions/v1beta1/namespaces/default/horizontalpodautoscalers/myrc
{
  "kind": "HorizontalPodAutoscaler",
  "apiVersion": "extensions/v1beta1",
...

curl localhost:8080/apis/autoscaling/v1/namespaces/default/horizontalpodautoscalers/myrc
{
  "kind": "HorizontalPodAutoscaler",
  "apiVersion": "autoscaling/v1",
...
```